File: CS/GridFilterButton/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Columns;
using DevExpress.Utils;
using DevExpress.XtraEditors.Drawing;
using DevExpress.Utils.Drawing;

namespace GridFilterButton
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        DataTable CreateTable()
        {
            DataTable table = new DataTable();
            DataRow dataRow;

            table.Columns.Add("Prod_NO", typeof(string));
            table.Columns.Add("Prod_Name", typeof(string));
            table.Columns.Add("Order_Date", typeof(string));
            table.Columns.Add("Quantity", typeof(string));

            dataRow = table.NewRow();
            dataRow["Prod_NO"] = "101";
            dataRow["Prod_Name"] = "Product1";
            dataRow["Order_Date"] = "12/06/2012";
            dataRow["Quantity"] = "50";
            table.Rows.Add(dataRow);

            dataRow = table.NewRow();
            dataRow["Prod_NO"] = "102";
            dataRow["Prod_Name"] = "Product2";
            dataRow["Order_Date"] = "15/06/2012";
            dataRow["Quantity"] = "70";
            table.Rows.Add(dataRow);

            dataRow = table.NewRow();
            dataRow["Prod_NO"] = "102";
            dataRow["Prod_Name"] = "Product2";
            dataRow["Order_Date"] = "15/06/2012";
            dataRow["Quantity"] = "70";
            table.Rows.Add(dataRow);

            dataRow = table.NewRow();
            dataRow["Prod_NO"] = "103";
            dataRow["Prod_Name"] = "Product3";
            dataRow["Order_Date"] = "18/06/2012";
            dataRow["Quantity"] = "30";
            table.Rows.Add(dataRow);

            dataRow = table.NewRow();
            dataRow["Prod_NO"] = "104";
            dataRow["Prod_Name"] = "Product4";
            dataRow["Order_Date"] = "25/06/2012";
            dataRow["Quantity"] = "20";
            table.Rows.Add(dataRow);

            return table;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            gridControl1.DataSource = CreateTable();
            foreach (GridColumn column in gridView1.Columns)
                column.AppearanceHeader.TextOptions.HAlignment = HorzAlignment.Center;
        }

        private void gridView1_CustomDrawColumnHeader(object sender, DevExpress.XtraGrid.Views.Grid.ColumnHeaderCustomDrawEventArgs e)
        {
            GridFilterButtonInfoArgs args = null;
            foreach (DrawElementInfo deInfo in e.Info.InnerElements)
                if (deInfo.ElementInfo.GetType() == typeof(GridFilterButtonInfoArgs))
                {
                    args = (GridFilterButtonInfoArgs)deInfo.ElementInfo;
                    break;
                }
            if (null == args) return;
            int x = e.Bounds.X + 5;
            int y = e.Bounds.Y + e.Bounds.Height / 2 - args.Bounds.Height / 2;
            args.Bounds = new Rectangle(new Point(x, y), args.Bounds.Size);
        }
    }
}

File: VB/GridFilterButton/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.Utils
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.Utils.Drawing

Namespace GridFilterButton
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'carsDBDataSet.Cars' table. You can move, or remove it, as needed.
			Me.carsTableAdapter.Fill(Me.carsDBDataSet.Cars)
			For Each column As GridColumn In gridView1.Columns
				column.AppearanceHeader.TextOptions.HAlignment = HorzAlignment.Center
			Next column
		End Sub

		Private Sub gridView1_CustomDrawColumnHeader(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.ColumnHeaderCustomDrawEventArgs) Handles gridView1.CustomDrawColumnHeader
			Dim args As GridFilterButtonInfoArgs = Nothing
			For Each deInfo As DrawElementInfo In e.Info.InnerElements
				If deInfo.ElementInfo.GetType() Is GetType(GridFilterButtonInfoArgs) Then
					args = CType(deInfo.ElementInfo, GridFilterButtonInfoArgs)
					Exit For
				End If
			Next deInfo
			If Nothing Is args Then
				Return
			End If
			Dim x As Integer = e.Bounds.X + 5
			Dim y As Integer = e.Bounds.Y + e.Bounds.Height / 2 - args.Bounds.Height / 2
			args.Bounds = New Rectangle(New Point(x, y), args.Bounds.Size)
		End Sub
	End Class
End Namespace